### ISSUE TITLE
Tap gesture event only supplies EventData; not GestureEventData

### ISSUE THREAD

**Issue by roblav96** [NativeScript/NativeScript#2775]
# tldr;
```xml
<Button text="Action" tap="{{ doAction }}" />
```
```typescript
doAction(args: GestureEventData) { // part of the page.bindingContext
	console.dir('args', args)
	console.dir('args.view', args.view) // << NULL
}
```
The `args` is only supplying `EventData`, **not** `GestureEventData`.

Modifying the `xml` element to include a `touch` binding will resolve the issue.
```xml
<Button text="Tap" tap="{{ onTap }}" touch="{{ onTouch }}" />
```
I don't believe I should need to include a `touch` binding as well.

---

When using `Angular Native` doing `(tap)="onTap($event)"`, the `$event` parameter is the correct `GestureEventData`.

### But with **`core`**,
```xml
<Button text="Action" tap="{{ doAction }}" />
```
```typescript
doAction(args: GestureEventData) { // part of the page.bindingContext
	console.dir('args', args)
	console.dir('args.view', args.view) // << NULL
}
```
The `args` is only supplying `EventData`, **not** `GestureEventData`.

### Steps to reproduce
```
tns create test --tsc
cd test
tns platform add ios
tns run ios
```

### Changing `onTap` to:
```typescript
public onTap(args: GestureEventData) {
	console.log('args', args)
	console.dir('args', args)
	console.log('args.view', args.view) // << undefined
	console.dir('args.view', args.view) // << undefined
}
```

### Console:
```shell
Sep 20 22:35:10 Robs-MBP test[6294]: CONSOLE LOG file:///app/main-view-model.js:27:20: args.view undefined
Sep 20 22:35:10 Robs-MBP test[6294]: CONSOLE LOG file:///app/main-view-model.js:28:20: args.view
```
Shouldn't `onTap` be emitting `GestureEventData`?

### Notes:
Doing
```xml
<Button text="Action" swipe="{{ onSwipe }}" />
```
with
```typescript
onSwipe(args: SwipeGestureEventData) {
	console.dir('args', args)
}
```
the args is correctly supplying the `SwipeGestureEventData`.

### `tns info`
```
┌──────────────────┬────────────────────────────┬────────────────┬─────────────┐
│ Component        │ Current version            │ Latest version │ Information │
│ nativescript     │ 2.3.0                      │ 2.3.0          │ Up to date  │
│ tns-core-modules │ 2.4.0-2016-09-20-4169      │ 2.3.0          │ Up to date  │
│ tns-android      │ 2.4.0-next-2016-09-14-1148 │ 2.3.0          │ Up to date  │
│ tns-ios          │ 2.4.0-2016-9-17-1          │ 2.3.0          │ Up to date  │
└──────────────────┴────────────────────────────┴────────────────┴─────────────┘
$ node -v
6.5.0
```

---

### Temporary Fix:
```xml
<Page navigatingTo="onNavigatingTo">
    <GridLayout class="main" rows="*">
        <StackLayout row="0" orientation="vertical">
            <Button id="action" text="Action" />
        </StackLayout>
    </GridLayout>
</Page>
```
```typescript
import * as application from "application"
import {Observable, EventData} from "data/observable"
import {GestureEventData, GestureTypes} from "ui/gestures"
import {Page, NavigatedData} from "ui/page"
import {View} from "ui/core/view"

export function onNavigatingTo(args: NavigatedData) {
	let page: Page = <Page>args.object
	page.bindingContext = new Index(
		page
	)
}
class Index extends Observable {
	constructor(
		private _page: Page
	) {
		super()
		let view: View = _page.getViewById('action')
		view.on(GestureTypes.tap, function(args: GestureEventData) {
			console.dump('args', args)
			console.dump('args.view', args.view) // NOT NULL :D
		})
	}
}
```
Grabbing the button view itself then binding the `GestureTypes.tap` event seems to do the trick, but shouldn't this not be necessary?

**Comment by roblav96** [comment]
### Edit 1
Seems that I can get the tap's `GestureEventData` if I **also apply** a `touch` gesture binding to the element.

### XML
```xml
<Button text="Tap" tap="{{ onTap }}" touch="{{ onTouch }}" />
```

**Comment by NickIliev** [comment]
Hey @roblav96 

The[ button component](https://docs.nativescript.org/cookbook/ui/button#responding-to-the-tap-event) has its own [tapEvent ](https://docs.nativescript.org/api-reference/classes/_ui_button_.button.html#tapevent) and indeed by design is working with EventData.

Here is how to use the button's tapEvent

```
  btn.on(Button.tapEvent, function (args: EventData) {
       console.log(args.object);
       console.log(args.eventName);
  });
```

**Comment by roblav96** [comment]
@NickIliev Yes that works great, but the correct parameters should be `GestureEventData` **NOT** `EventData`.

Like this is how it's supposed to work, but doesn't:
```typescript
  btn.on(Button.tapEvent, function (args: GestureEventData) {
       console.log(args.object)
       console.log(args.eventName)
       console.log(args.view) // << NULL
       console.log(args.ios) // << NULL
       console.log(args.android) // << NULL
  })
```

**Comment by enchev** [comment]
Hey @roblav96,

The Button is special case since there is already `tap` event with no gestures needed.

**Comment by roblav96** [comment]
Ah, ok. That makes sense. Thank you!
